File: api/src/main/java/com/api/app/emum/PAY003.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 결제상태코드: payStatusCode
 * */
@Getter
@AllArgsConstructor
public enum PAY003 implements CommonCode {
    PAYMENT_PENDING("001", "결제대기", 1, "", ""),
    PAYMENT_COMPLETED("002", "결제완료", 2, "", ""),
    PAYMENT_CANCELLED("003", "결제취소", 3, "", ""),
    REFUND_RECEIVED("101", "환불접수", 4, "", ""),
    REFUND_COMPLETED("102", "환불완료", 5, "", ""),
    REFUND_CANCELLED("103", "환불접수취소", 6, "", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static PAY003 findByCode(String code) {
        return CommonCodeUtil.findByCode(PAY003.class, code);
    }
}

File: api/monolith/src/main/kotlin/com/api/app/emum/PAY003.kt
package com.api.app.emum

/** 결제상태코드: payStatusCode */
enum class PAY003(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    PAYMENT_PENDING("001", "결제대기", 1, "", ""),
    PAYMENT_COMPLETED("002", "결제완료", 2, "", ""),
    PAYMENT_CANCELLED("003", "결제취소", 3, "", ""),
    ;

    companion object {
        fun findByCode(code: String): PAY003? = CommonCodeUtil.findByCode(PAY003::class.java, code)
    }
}
